app/src/main/kotlin/com/flxrs/dankchat/channels/ChannelsAdapter.kt

```diff
@@ -21,14 +25,11 @@ class ChannelsAdapter(private val onEditChannel: (String, String?) -> Unit) : Li
         val dankChatPreferences = DankChatPreferenceStore(holder.itemView.context)
         val channel = getItem(position)
         val renamedChannel = dankChatPreferences.getRenamedChannel(channel)
-        when (renamedChannel) {
-            null, channel -> {
-                channelText.text = channel
-                ogChannelText.text = ""
-            }
-            else ->{
-                channelText.text = renamedChannel
-                ogChannelText.text = channel
+        channelText.text = buildSpannedString {
+            append(renamedChannel ?: channel)
+            renamedChannel?.let {
+                val channelColor = ColorUtils.setAlphaComponent(channelText.currentTextColor, 128)
+                if (it != channel) color(channelColor) { italic { append(" $channel") } }
```

Comment:
Should be `append(" $it`"). Also we can move the if out and use Kotlins util functions `takeIF`:
```kotlin
renamedChannel
    ?.takeIf { it != channel }
    ?.let {
        val channelColor = ColorUtils.setAlphaComponent(channelText.currentTextColor, 128)
        color(channelColor) { italic { append(" $it") } }
    }
```

Reply:
``It`` is equal to the channel rename If one exists. But we want to display the original channel name in italic no?

Reply:
Yeah sorry I had a brainfart. Should be `channel` of course

Reply:
But I would still like the `takeIf` solution :)